projects/centre_console/inc/centre_console_fault_reason.h

```diff
@@ -5,33 +5,35 @@
 #include "stdint.h"
 
 typedef enum {
-  POWER_MAIN_FAULT_REASON_ACK_FAIL = 0,
-  NUM_POWER_MAIN_FAULT_REASONS
-} PowerMainFaultReason;
+  DRIVE_FSM_STATE_MACHINE = 0,
+  POWER_MAIN_SEQUENCE_STATE_MACHINE,
+  POWER_OFF_SEQUENCE_STATE_MACHINE,
+  POWER_AUX_SEQUENCE_STATE_MACHINE,
+  NUM_STATE_MACHINES,
+} StateMachine;
 
-typedef enum {
-  CENTRE_CONSOLE_FAULT_REASON_BMS_CARRIER_FAILED_TO_TURN_ON = 0,
-  CENTRE_CONSOLE_FAULT_REASON_AUX_VOLTAGE_LOW,
-  CENTRE_CONSOLE_FAULT_REASON_MCI_FAILED_TO_TURN_ON,
-  CENTRE_CONSOLE_FAULT_REASON_BATTERY_RELAY_FAILED_TO_CLOSE,
-  CENTRE_CONSOLE_FAULT_REASON_DCDC_NOT_SWITCHED,
-  CENTRE_CONSOLE_FAULT_REASON_PRECHARGE_NOT_INITIATED,
-  CENTRE_CONSOLE_FAULT_REASON_MOTOR_RELAY_FAILED_TO_CLOSE,
-  NUM_CENTRE_CONSOLE_FAULT_REASONS
-} CentreConsoleFaultReason;
+// Used as the event data for the CENTRE_CONSOLE_POWER_EVENT_FAULT, and
+// DRIVE_FSM_INPUT_EVENT_FAULT events.
+typedef union StateTransitionFault {
+  struct {
+    StateMachine state_machine : 3;
+    uint8_t fault_reason : 8;
```

Comment:
Actually, shouldn't this be `DriveFsmFaultReason` that way you don't have to use that yucky `((DriveFsmFaultReason){ ... }).raw` stuff?